### ISSUE TITLE
Bloody Fun Day: Tutorial text formatting and Critter spawn RNG

### ISSUE THREAD

**Issue by waspennator** [ruffle-rs/ruffle#12387]
### Describe the bug

https://www.newgrounds.com/portal/view/486608?emulate=flash

1: The text boxes that spawn during the tutorial don't seem to be formatted correctly, resulting in the text being hard to read.

2: When playing any 5 level or infinite (does not happen on tutorial), the critters all seem to spawn in as one singular color.

1:
![Screenshot_20230728_183154_Chrome](https://github.com/ruffle-rs/ruffle/assets/78768499/56a7e871-4658-4ef9-96b4-4adc4815e4b3)

2:
![Screenshot_20230728_182143_Chrome](https://github.com/ruffle-rs/ruffle/assets/78768499/81f6610b-3106-4e77-a866-718127b6483f)

### Expected behavior

The text should be properly formatted in the tutorial and the critter RNG shouldn't just spawn em all as the same color in 5 level or infinite mode.

### Affected platform

Browser's extension

### Operating system

Android 13

### Browser

Chrome 115

**Comment by Aaron1011** [comment]
The critter spawn issue is caused by a bug in how we handle integer keys in a `Dictionary`.

This code:

```actionscript
import flash.utils.Dictionary;

var dict = new Dictionary();
dict[100] = "Test";
dict[1.2] = "Hello";

for (var key in dict) {
	trace("Key: " + key + " type: " + typeof(key));
}
```

prints this under flash:

```
Key: 100 type: number
Key: 1.2 type: string
```

but this under Ruffle:

```
Key: 100 type: string
Key: 1.2 type: string
```

The SWF tries to convert the key to a `uint,` which gives us `null` (and later `0`) when we have a string instead of a `number`.

**Comment by waspennator** [comment]
Still happening as of latest nightly.

**Comment by waspennator** [comment]
The 1st issue seems to be fixed now, but the critter rng is still borked.

**Comment by Lord-McSweeney** [comment]
Fixed by #14721
